vcstool/clients/vcs_base.py

```diff
@@ -93,7 +113,12 @@ def load_url(url, retry=2, retry_period=1, timeout=10):
     try:
         fh = urlopen(url, timeout=timeout)
     except HTTPError as e:
-        if e.code == 503 and retry:
+        if e.code in (401, 404):
+            # Try again, but with authentication
+            fh = _authenticated_urlopen(url, timeout=timeout)
```

Comment:
Does the sibling function `test_url` not need the same logic?

Reply:
Good catch, I missed that feature entirely :blush: .

Reply:
Added in f3286273b5150016e991a51173e3ab6cfa2e13fd.